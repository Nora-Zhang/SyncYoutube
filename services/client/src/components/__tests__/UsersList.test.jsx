import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import UsersList from '../UsersList';

const users = [
  {
    'active': true,
    'email': 'bot1@botland.com',
    'id': 1,
    'username': 'bot1'
  },
  {
    'active': true,
    'email': 'bot2@botland.com',
    'id': 2,
    'username': 'bot2'
  }
];

test('Test UsersList renders correctly', () => {
  const wrapper = shallow(<UsersList users={users} />);
  const element = wrapper.find('h4');
  expect(element.length).toBe(2);
  expect(element.get(0).props.children).toBe('bot1');
})

test('UsersList renders a snapshot properly', () => {
  const tree = renderer.create(<UsersList users={users} />).toJSON();
  expect(tree).toMatchSnapshot();
})