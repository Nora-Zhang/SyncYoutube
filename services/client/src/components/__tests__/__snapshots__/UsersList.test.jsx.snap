// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList renders a snapshot properly 1`] = `
<div>
  <h4
    className="box title is-4"
  >
    bot1
  </h4>
  <h4
    className="box title is-4"
  >
    bot2
  </h4>
</div>
`;
